src/KafkaFlow.Serializer.JsonCore/JsonCoreSerializer.cs

```diff
@@ -27,21 +29,22 @@ public JsonCoreSerializer()
         {
         }
 
-        /// <summary>Serializes the message</summary>
-        /// <param name="message">The message to be serialized</param>
-        /// <returns>A UTF8 JSON string</returns>
-        public byte[] Serialize(object message)
+        /// <inheritdoc/>
+        public Task SerializeAsync(object message, Stream output, ISerializationContext context)
         {
-            return JsonSerializer.SerializeToUtf8Bytes(message, this.options);
+            using var writer = new Utf8JsonWriter(output);
+
+            JsonSerializer.Serialize(writer, message, this.options);
```

Comment:
Considering that is a memory operation, is faster to use the sync method than the async. The async option was used in the `DeserializeAsync` because there is no overload to use Streams with the sync method in the System.Text.Json 😢